cypress/e2e/1-general/status_page_channels.cy.js

```diff
@@ -0,0 +1,298 @@
+          for (let i = 0; i < channelData.length; i++) {
+            cy.setChannelOutput(channelData[i].slot, 0); // turn each channel off with the api
+          }
+        });
+        // now turn on with the dom
+        cy.get('.filter-channels-section button.is-active'); // wait for DOM to update
+        cy.liveData().its('body').then(liveData => {
+          const channelData = liveData.channels;
+          for (let i = 0; i < channelData.length; i++) {
+            cy.channelTableRow(i).within(async () => {
+              cy.get(`td:nth-child(${4 + ltgOffset}) .react-toggle`, { timeout: 7000 }).should('not.have.class',
+                'react-toggle--checked');
+              cy.get(`td:nth-child(${4 + ltgOffset}) .react-toggle`).click();
+              cy.get(`td:nth-child(${4 + ltgOffset}) .react-toggle`).should('have.class',
+                'react-toggle--checked');
+            });
+          };
+        });
+        cy.wait(2500); // system settling time
+        // fetch live data again, it should reflect the new state ON
+        cy.liveData().its('body').then(liveData => {
+          const channelData = liveData.channels;
+          const rowCount = channelData.length;
+          for (let i = 0; i < rowCount; i++) {
+            const mode = channelData[i].mode
+            cy.channelTableRow(i).within( () => {
+              expect(mode).to.not.equal(MODES.OUTPUT_DISABLED);
+            });
+          }; 
+        });   
+      });
+    });
+  });
+
+
+  describe('- live ID control', () => {
+    beforeEach(() => {
+      cy.get('.filter-channels-section button').then(($btn) => {
+        if ($btn.text() === 'ALL CHANNELS') { $btn.click(); } // only click if present
+      });
+      cy.get('.filter-channels-section button.is-active'); // wait for DOM to update
+    });
+
+    context('when channels live ID are OFF', () => {
+      beforeEach(() => {
+        // ensure each channel is in a known state (live ID off)
+        cy.log('turning live ID off for all slot');
+        cy.liveData().its('body').then(liveData => {
+          const channelData = liveData.channels;
+          for (let i = 0; i < channelData.length; i++) {
+            cy.reqSetliveID(channelData[i].slot, 0); // set each channels live ID off
+          }
+        });
+      });
+      it('can turn each channel live ID ON with liveData', () => {
+        cy.liveData().its('body').then(liveData => {
+          const channelData = liveData.channels;
+          // toggle each channel row
+          for (let i = 0; i < channelData.length; i++) {
+            cy.channelTableRow(i).within(async () => {
+              cy.get(`td:nth-child(${5 + ltgOffset}) .react-toggle`,
+                { timeout: 7000 }).should('not.have.class', 'react-toggle--checked');
+              cy.get(`td:nth-child(${5 + ltgOffset}) .react-toggle`).click();
+              cy.get(`td:nth-child(${5 + ltgOffset}) .react-toggle`).should('have.class', 'react-toggle--checked');
```

Comment:
Because we're repeating this ugly selector so many times you could create a helper function like
```
function getLiveIdButton() {
  return cy.get(`td:nth-child(${5 + ltgOffset}) .react-toggle`)
}
```
if you just define the function in this file then it has access to 'ltgOffset' so you don't even have to pass it as a param. Then just:

```suggestion
              getLiveIdButton().should('not.have.class', 'react-toggle--checked');
              getLiveIdButton().click();
              getLiveIdButton().should('have.class', 'react-toggle--checked');
```